### PR TITLE
refactor: notification.ts

### PR DESCRIPTION
**Pull request by trim21**

### CODE REVIEW

**Review by aldy505 — COMMENTED** [review]
LGTM

File: src/minio.js [hunk 1]
```diff
@@ -54,15 +54,18 @@ import {
   uriResourceEscape,
 } from './internal/helper.ts'
 import { PostPolicy } from './internal/post-policy.ts'
-import { NotificationConfig, NotificationPoller } from './notification.js'
+
+import { NotificationConfig, NotificationPoller } from './notification.ts'
+import { ObjectUploader } from './object-uploader.js'
```
Comment (trim21) [inline src/minio.js:59]:
```suggestion
```

File: src/minio.js [hunk 2]
```diff
@@ -54,15 +54,17 @@ import {
   uriResourceEscape,
 } from './internal/helper.ts'
 import { PostPolicy } from './internal/post-policy.ts'
-import { NotificationConfig, NotificationPoller } from './notification.js'
+
```
Comment (trim21) [inline src/minio.js:57]:
```suggestion
```

File: src/minio.js [hunk 3]
```diff
@@ -54,15 +54,16 @@ import {
   uriResourceEscape,
 } from './internal/helper.ts'
 import { PostPolicy } from './internal/post-policy.ts'
-import { NotificationConfig, NotificationPoller } from './notification.js'
+import { NotificationConfig, NotificationPoller } from './notification.ts'
+
```
Comment (trim21) [inline src/minio.js:58]:
```suggestion
```

File: src/minio.d.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // imported from https://github.com/DefinitelyTyped/DefinitelyTyped/blob/93cfb0ec069731dcdfc31464788613f7cddb8192/types/minio/index.d.ts
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
-import { EventEmitter } from 'node:events'
+import type { Readable as ReadableStream } from 'node:stream'
```
Comment (trim21) [inline src/minio.d.ts:4]:
```suggestion
import { EventEmitter } from 'node:events'
```

**Comment by trim21** [comment]
@kaankabalak

**Comment by prakashsvmx** [comment]
@trim21 , could you please rename `examples/minio/listen-bucket-notification.js`  to `examples/minio/listen-bucket-notification.mjs`  ?

**Comment by trim21** [comment]
> @trim21 , could you please rename `examples/minio/listen-bucket-notification.js` to `examples/minio/listen-bucket-notification.mjs` ?

This file is not touched

**Comment by aldy505** [comment]
> @trim21 , could you please rename `examples/minio/listen-bucket-notification.js` to `examples/minio/listen-bucket-notification.mjs` ?

@prakashsvmx I think changing the example is better be done at https://github.com/minio/minio-js/pull/1236 or in another separate PR. We should be more focused on refactoring the project to Typescript first. Then after some refactored code got merged, we can work on the examples, then create a new release. It should be faster than changing the example alongside the code, while having to go back and forth resolving merge conflict everytime a certain PR got merged.

**Review by prakashsvmx — APPROVED** [review]

**Comment by trim21** [comment]
@kaankabalak

**Review by kaankabalak — APPROVED** [review]

**Pull request merged**